### ISSUE TITLE
Resend messages that are not in the history after reconnect

### ISSUE THREAD

**Issue by ChristophHaag** [EionRobb/pidgin-opensteamworks#140]
Sometimes the plugin loses the connection to steam and doesn't immediately notice - perhaps the keepalive messages are not sent often enough? Anyway, the problem with this is that the status of the account remains "online" for a while. You can write messages while the plugin is in this limbo and you don't get any notice that sending the message fails. But when you open the real steam client and look at the history, those messages are missing.

The feature request is that those messages are resent after the reconnect happens.
I.e.
1. download the history
2. compare with the local history
3. resend our own messages that are not in the history (and that aren't too old)

**Comment by xpander69** [comment]
would be nice to have indeed.
this is really nasty issue. have to check steam client all the time if there was something that wasnt sent through pidgin

**Comment by Spanguole** [comment]
Indeed, this is very annoying and makes it so that running the steam client together with pidgin, just so messages would not get silently dropped and it would be possible to see which have been.  
Which, in part, defeats the purpose of having an alternative chat client...

**Comment by LemonadeCellar** [comment]
I know this issue is old, but I had the same problem for as long as I've been using this plugin. I run Pidgin with the plugin and desktop Steam in parallel, and sometimes Pidgin disconnects and doesn't reconnect for at least 10 minutes while the desktop Steam client continues to work fine and send/receive messages like normal, so then bits of conversations end up missing from the Pidgin logs.
